components/ThoughtList/index.js

```diff
@@ -1,158 +1,202 @@
-import React, { Component } from 'react'
-import { View, TouchableOpacity, TextInput, Dimensions, StyleSheet } from 'react-native'
-import { MaterialIcons, Octicons, Ionicons } from '@expo/vector-icons'
-const { width, height } = Dimensions.get('window');
+import React, { useState } from 'react';
+import { View, Text, TouchableOpacity, TextInput } from 'react-native'
+import { Container, Header, Content, Left } from 'native-base';
+import styles from './styles'
+import { Dot, Add, Hide, Show, Menu } from '../StyledIcon'
+import { addNewThought } from '../../src/actions/ThoughtAction'
+import { connect } from 'react-redux'
 
-export default class ThoughtList extends Component {
-    constructor(props) {
-        super(props)
-        this.state = {
-            thoughtsList: [{}],
-            thought: '',
+export default connect(({thoughtsList})=>({thoughtsList}), ({ addNewThought }))(({ ...props }) => {
+  console.log(props )
+  const [thought, setThought] = useState('');
+  const [focusedThought, setFocusedThought] = useState('');
+  const [deleteItem, setDeleteItem] = useState(false);
+  let tempList = []
```

Comment:
The linter must not be set up properly, because `let` statements are not allowed.

Reply:
There is no rule such as no let in lint.

There are rules for:

no var
prefer const over let

let should not be to used when variable value is not going to be change.

Please refer this for clarification:

https://eslint.org/docs/rules/no-var

https://eslint.org/docs/rules/prefer-const

https://palantir.github.io/tslint/rules/prefer-const/

Reply:
It's called `fp/no-let` and it is part of the `fp` eslint plugin, which you can see in the `.eslintrc.json`.